src/main/java/seedu/address/logic/recommender/Recommender.java

```diff
@@ -0,0 +1,134 @@
+package seedu.address.logic.recommender;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.recommender.location.LocationRecommender;
+import seedu.address.logic.recommender.location.LocationTracker;
+import seedu.address.model.Model;
+import seedu.address.model.location.Location;
+import seedu.address.model.person.ContactIndex;
+import seedu.address.model.recommendation.Recommendation;
+import seedu.address.model.time.HourBlock;
+import seedu.address.model.time.TimePeriod;
+import seedu.address.model.timingrecommender.TimingRecommender;
+
+/**
+ * Recommends meetup times and locations.
+ */
+public class Recommender {
+
+    private static final Logger logger = LogsCenter.getLogger(Recommender.class);
+    private static final int RECOMMENDATION_LIMIT = 20;
+    private final LocationRecommender lr;
+    private final TimingRecommender sc;
```

Comment:
Do you think this needs renaming? lr I can understand, but why sc for TimingRecommender?

Reply:
OK :)

Reply:
Fixed to have the full names instead.